examples/template-angular-ts/src/app/app.component.html

```diff
@@ -0,0 +1,13 @@
+<ng-container *ngIf="state$ | async as state">
```

Comment:
can `state` ever be undefined here? i might not understand the syntax but I wonder why do we need `ngIf` here. Ideally, the state should be provided synchronously to the template

Reply:
Yes, you're right. I like using observables on my apps because I use `ChangeDetectionStrategy.OnPush` on all of my components and the `async` pipe marks the view for check behind the scenes whenever the observable emits a value.

I will change it to use the sync version as this is a basic example.

Reply:
> <ng-container *ngIf="state$ | async as state">

So this is just a standard way of unwrapping a non-nullable observable?

Reply:
That's right

Reply:
Roger that, doesn't it get annoying when u have a couple of observables in a component? Or do you `combineLatest` them then, to unwrap and destructure in the template with a single `ng-container`? 🤔

Reply:
I do just that, `combineLatest` and single `ng-container`